pkg/rpc/client/client.go

```diff
@@ -64,32 +61,34 @@ func New(ctx context.Context, endpoint string, opts Options) (*Client, error) {
 		return nil, err
 	}
 
+	if opts.DialTimeout == 0 {
```

Comment:
I believe `<= 0` is more appropriate. E.g. if underlying machinery decides to handle negative values with it's own default value.